### PR TITLE
Add throw if white space

### PR DESCRIPTION
**Pull request by MattLangers**
Realized there is already a throw if whitespace, so made this public and added a return early if a null is passed to it. also took the opportunity to rename to System1Group.Lib.CoreUtils

### CODE REVIEW

File: CoreUtils.Tests/ThrowHelpers/ReturnParameter/OrThrowIfWhiteSpace.Tests.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        }
+
+        [Test]
+        public void ReturnParameter_OrThrowIfWhiteSpace_Null()
+        {
+            Assert.DoesNotThrow(() => ReturnParameter.OrThrowIfWhitespace((string)null, "test"));
+        }
+
+        [Test]
+        public void ReturnParameter_OrThrowIfNull_Null()
```
Comment (K1Z4) [inline CoreUtils.Tests/ThrowHelpers/ReturnParameter/OrThrowIfWhiteSpace.Tests.cs]: Looks like the function name does not match the test

File: CoreUtils.Tests/ThrowHelpers/Throw/IfWhiteSpace.Tests.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public class Throw_IfWhiteSpace_Tests
+    {
+        [Test]
+        public void Throw_IfWhitespace_Ok()
+        {
+            Assert.DoesNotThrow(() => Throw.IfWhitespace("hello", "TestParam"));
+        }
+
+        [Test]
+        public void Throw_IfNull_Null()
```
Comment (K1Z4) [inline CoreUtils.Tests/ThrowHelpers/Throw/IfWhiteSpace.Tests.cs]: Looks like these bottom two tests also have the wrong method name in the test names

Comment (MattLangers) [reply CoreUtils.Tests/ThrowHelpers/Throw/IfWhiteSpace.Tests.cs]: Good spot, thanks

File: CoreUtils.Tests/ThrowHelpers/ReturnParameter/OrThrowIfWhiteSpace.Tests.cs (hunk 1 above)
Comment (MattLangers) [reply CoreUtils.Tests/ThrowHelpers/ReturnParameter/OrThrowIfWhiteSpace.Tests.cs]: Good spot, thanks

**Pull request merged**